### PR TITLE
Add Microsoft.Build.16.10.0 and Microsoft.Build.Tasks.Core.16.10.0

### PR DESCRIPTION
**Pull request by mthalman**
The latest 8.0 Preview 2 build introduced the following prebuilts:

* Microsoft.Build.16.10.0
* Microsoft.Build.Framework.16.10.0
* Microsoft.Build.Tasks.Core.16.10.0
* Microsoft.Build.Utilities.Core.16.10.0
* System.Configuration.ConfigurationManager.4.7.0
* System.Security.Cryptography.ProtectedData.4.7.0
* System.Security.Cryptography.Xml.4.7.0

These changes add those packages to SBRP. All of these were included as a result of running the generate script on two packages: Microsoft.Build.16.10.0 and Microsoft.Build.Tasks.Core.16.10.0.

One issue that was run into was with the changes from https://github.com/dotnet/source-build-reference-packages/pull/547. It resulted in the following error:

```
ReferencePackageSourceGenerator -> Generating package source for System.Reflection.Emit (v4.3.0)...
  Determining projects to restore...
  Restored /mnt/c/repos/source-build-reference-packages/src/referencePackageSourceGenerator/ReferencePackageSourceGenerator.proj (in 20 ms).
  1 of 2 projects are up-to-date for restore.
  Encountered package 'system.reflection.emit.4.3.0' with excluded target framework 'xamarinmac20'.
  Encountered package 'system.reflection.emit.4.3.0' with excluded target framework 'netcore50'.
  Encountered package 'system.reflection.emit.4.3.0' with excluded target framework 'xamarinios10'.
  Encountered package 'system.reflection.emit.4.3.0' with excluded target framework 'xamarintvos10'.
Encountered package 'system.reflection.emit.4.3.0' with excluded target framework 'xamarinwatchos10'.
Determining projects to restore...
All projects are up-to-date for restore.
/mnt/c/repos/source-build-reference-packages/.dotnet/sdk/8.0.100-preview.1.23115.2/Microsoft.Common.CurrentVersion.targets(1229,5): error MSB3644: The reference assemblies for MonoTouch,Version=v1.0 were not found. To resolve this, install the Developer Pack (SDK/Targeting Pack) for this framework version or retarget your application. You can download .NET Framework Developer Packs at https://aka.ms/msbuild/developerpacks [/mnt/c/repos/source-build-reference-packages/artifacts/bin/ReferencePackageSourceGenerator/system.reflection.emit/4.3.0/system.reflection.emit.4.3.0.csproj::TargetFramework=monotouch10]
```

This was resolved by fixing the ExcludedTargetFrameworks property for monotouch.

### CODE REVIEW

File: src/referencePackages/src/microsoft.bcl.asyncinterfaces/1.1.0/Microsoft.Bcl.AsyncInterfaces.1.1.0.csproj [hunk 1]
```diff
@@ -1,8 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
-  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))/dir.props" />
```
Comment (lbussell) [inline src/referencePackages/src/microsoft.bcl.asyncinterfaces/1.1.0/Microsoft.Bcl.AsyncInterfaces.1.1.0.csproj:2]: I think in general we don't want to check-in changes to projects that have already been added to SBRP, since they usually have required some work to get to a buildable state - unless there's a particular reason these have all been touched. I think this applies to most other projects in this PR.

**Review by lbussell — CHANGES_REQUESTED** [review]

File: src/referencePackages/src/system.drawing.common/4.7.0/System.Drawing.Common.4.7.0.csproj [hunk 2]
```diff
@@ -0,0 +1,31 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (lbussell) [inline src/referencePackages/src/system.drawing.common/4.7.0/System.Drawing.Common.4.7.0.csproj:1]: Many `4.7.0` versions of packages have known vulnerabilities in them. Even though we don't include any of the vulnerable source code, they trigger CG alerts. In the instances where we've updated these projects in the past, you should use the checked-in updated version and update the reference from the dependent project. See here: https://github.com/dotnet/source-build-reference-packages/pulls?q=is%3Apr+4.7.0

**Review by lbussell — CHANGES_REQUESTED** [review]

File: src/referencePackages/src/microsoft.build.framework/16.10.0/Microsoft.Build.Framework.16.10.0.csproj [hunk 3]
```diff
@@ -25,6 +25,7 @@
 
   <ItemGroup Condition=" '$(TargetFramework)' == 'net472' ">
     <PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies.net472" Version="1.0.2" />
+    <PackageReference Include="System.Runtime" Version="4.3.0" />
```
[the remark this answers is not in the document]
Comment (mthalman) [reply src/referencePackages/src/microsoft.build.framework/16.10.0/Microsoft.Build.Framework.16.10.0.csproj:28]: Without it, I get a bunch of these errors:

```
error CS0234: The type or namespace name 'ComponentModel' does not exist in the name
space 'System' (are you missing an assembly reference?) [C:\repos\source-build-reference-packages\src\referencePackages\src\microsoft.build.framework\16.10.0\Microsoft.Build.Framework.16.10.0.csproj::TargetFramework=net472]
```

[the remark this answers is not in the document]
Comment (ViktorHofer) [reply src/referencePackages/src/microsoft.build.framework/16.10.0/Microsoft.Build.Framework.16.10.0.csproj:28]: Can you please post the exact errors? I don't think that a package reference to `System.Runtime` is correct here.

File: src/referencePackages/src/microsoft.build/16.10.0/Microsoft.Build.16.10.0.csproj [hunk 4]
```diff
@@ -27,7 +27,6 @@
     <PackageReference Include="System.Memory" Version="4.5.4" />
     <PackageReference Include="System.Text.Json" Version="4.7.0" />
     <PackageReference Include="System.Threading.Tasks.Dataflow" Version="4.9.0" />
-    <PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies.net472" Version="1.0.2" />
```
Comment (ViktorHofer) [inline src/referencePackages/src/microsoft.build/16.10.0/Microsoft.Build.16.10.0.csproj]: For my own understanding and to improve the existing tooling, why did you remove this package reference?

**Comment by ViktorHofer** [comment]
@mthalman as the new GenAPI and SBRP generator are a complete rewrite, would you mind summarizing the issues that you observed while generating these packages? Any feedback is welcome.

File: src/referencePackages/src/microsoft.build.framework/16.10.0/Microsoft.Build.Framework.16.10.0.csproj (hunk 3 above)
[the remark this answers is not in the document]
Comment (mthalman) [reply src/referencePackages/src/microsoft.build.framework/16.10.0/Microsoft.Build.Framework.16.10.0.csproj:28]: The failed build before adding this change is here: https://dev.azure.com/dnceng-public/public/_build/results?buildId=194813&view=logs&jobId=2f0d093c-1064-5c86-fc5b-b7b1eca8e66a&j=2f0d093c-1064-5c86-fc5b-b7b1eca8e66a&t=d2b639a6-cb19-5f1f-66fd-8047f66b3026.

Note that this was added to the already existing 16.8.0 version of this package:

https://github.com/dotnet/source-build-reference-packages/blob/33b9ba860dee1f4ae2bc3fa727e5fe64a9c6466c/src/referencePackages/src/microsoft.build.framework/16.8.0/Microsoft.Build.Framework.16.8.0.csproj#L35

File: src/referencePackages/src/microsoft.build/16.10.0/Microsoft.Build.16.10.0.csproj (hunk 4 above)
Comment (mthalman) [reply src/referencePackages/src/microsoft.build/16.10.0/Microsoft.Build.16.10.0.csproj]: We're troubleshooting trying to find a way to get the build to work. There's an issue in the build of Microsoft.Build where it's loading the Microsoft.Build.Framework assembly from microsoft.netframework.referenceassemblies.net472 instead of from the Microsoft.Build.Framework package that was just built.

File: src/referencePackages/src/microsoft.build.framework/16.10.0/Microsoft.Build.Framework.16.10.0.csproj (hunk 3 above)
[the remark this answers is not in the document]
Comment (ViktorHofer) [reply src/referencePackages/src/microsoft.build.framework/16.10.0/Microsoft.Build.Framework.16.10.0.csproj:28]: Thanks. Looking at the very first error from the build:

>   /__w/1/s/artifacts/source-build/self/src/src/referencePackages/src/microsoft.build.framework/16.10.0/lib/net472/Microsoft.Build.Framework.cs(1144,51): error CS0234: The type or namespace name 'ComponentModel' does not exist in the namespace 'System' (are you missing an assembly reference?) [/__w/1/s/artifacts/source-build/self/src/src/referencePackages/src/microsoft.build.framework/16.10.0/Microsoft.Build.Framework.16.10.0.csproj::TargetFramework=net472] [/__w/1/s/.packages/microsoft.dotnet.arcade.sdk/8.0.0-beta.23153.1/tools/Build.proj]

That API is exposed in .NET Framework in System.dll: https://referencesource.microsoft.com/#System/compmod/system/componentmodel/ISupportInitialize.cs,20

Does adding a reference to "System.dll" not fix the issue?

**Comment by mthalman** [comment]
Abandoning this because we're defining these as prebuilts instead. The main issue that was encountered was that the build of the Microsoft.Build package was using the Microsoft.Build.Framework assembly from the net472 targeting pack instead of the newly built Microsoft.Build.Framework package. This is only an issue for the net472 TFM so this should be resolved when TFM filtering is online.
